docs: Flex manual, improve SSH instructions (#18854)
OpenTrons/opentrons-api: docs/flex-manual/docs/software-operation.md

Flex *will not* retain information about more than 20 runs on the robot. Proceeding to the Run Setup screen generates an entry in the list and counts towards the maximum of 20 runs, even if you never begin the protocol.

> ## Command Line Operation Over SSH

Reviewer: Sentence case headers here and throughout.

Author: Changing to sentence case headers

> You can work with your Flex through a Secure Shell (SSH) terminal connection. Terminal access lets you run protocols directly from the command line or perform advanced tasks, such as customizing the Python environment on the robot. Protocols that reference external files on disk (apart from custom labware definition files) must be run from the command line.

Reviewer: Let's preserve the link here, even if it goes to a fairly information-light section of the API docs. (We could move or replicate that info here, but I don't think that's necessary for this PR.)
```suggestion
You can work with your Flex through a Secure Shell (SSH) terminal connection. Terminal access lets you [run protocols directly from the command line](https://docs.opentrons.com/v2/new_advanced_running.html#command-line) or perform advanced tasks, such as customizing the Python environment on the robot. Protocols that reference external files on disk (apart from custom labware definition files) must be run from the command line.
```

!!!Note
    - SSH keys are required before you can connect to Flex and issue commands from a terminal.

>     - If you're unable to use a Wi-Fi network for SSH, see Hardwired SSH Connections below.

Reviewer: Let's link to that section here.
```suggestion
    - If you're unable to use a Wi-Fi network for SSH, see [Hardwired SSH Connections][hardwired-ssh-connections] below.
```

1. Copy the `robot_key.pub` file to the root of a USB-A flash drive. You will use this USB drive (and the saved key) for SSH authentication to the robot.

    !!!note
        The flash drive must have a single partition formatted with a file system readable by the embedded Linux system on Flex. FAT32, NTFS, and ext4 file systems are supported. The macOS HFS+ and APFS file systems are not. macOS can read and write to FAT-formatted drives.

1. Eject the USB drive.

### Making an SSH Connection

To make an SSH connection:

1. Insert the USB drive that holds the SSH key created earlier into a USB port on your Flex.

1. On your computer, open a terminal window and type the commands shown below. Replace `ROBOT_IP` with the IP address of your Flex.

    ```
    curl \
    --location --request POST \
    'http://ROBOT_IP:31950/server/ssh_keys/from_local'
    ```
    The command is successful when you see a response message that indicates a new key was added.

1. After adding the key, type the command shown below. Replace `ROBOT_IP` with the IP address of your Flex.

    ```
    ssh -i robot_key root@ROBOT_IP
    ```

1. Type the passphrase you set when creating the SSH key.

> When an SSH connection is successful, the terminal command prompt changes to `root@Flex` followed by the serial number of your robot (e.g., `root@FLXA1020231007001:~#`). You can now interact with the robot via the terminal window.

Reviewer: Slight change to keep this consistent, since the example prompt doesn't include the string `Flex` (only the `FLX` that's part of the serial number).
```suggestion
When an SSH connection is successful, the terminal command prompt changes to `root@` followed by the serial number of your robot (e.g., `root@FLXA1020231007001:~#`). You can now interact with the robot via the terminal window.
```

### Hardwired SSH Connections

A hardwired connection uses an Ethernet cable to connect and transmit data directly between your computer and Flex. This is a secure alternative for SSH access in situations where network policies prevent you from making a wireless connection to the robot.

!!!note

>     The hardwired SSH procedure requires familiarity with working in a terminal window and setting a static IP address. You may want to ask for help from your IT support team before proceeding.

Reviewer: I would recommend removing "familiarity with working in a terminal window" since all of the setup instructions above require that too.

Author: Removing quoted text. Will leave in same section. Will use:

> The hardwired SSH procedure requires assigning a static IP address to the robot. You may want to ask for help from your IT support team before proceeding.

#### Physical Connection

> Connect a laptop to the robot using an Ethernet cable. If your laptop has a built-in RJ-45 Ethernet port, plug one end into the computer and connect the other end to the Ethernet port on the robot. If you're using a laptop without an Ethernet port, use an adapter with an Ethernet port to make this connection.

Reviewer: I think we should replace "laptop" with "computer" throughout. There's no reason you need to use a laptop for this, except the convenience of being able to bring it close to the Flex.

Author: Changing to "computer"

When disconnected from a network, your Flex will assign itself an IP address and subnet mask. You'll need this information to set a static address on your computer within the same IP address range and subnet as your Flex.

#### Finding the Robot's IP Address

You can get the IP address range and subnet mask from the robot after connecting it to your computer and checking the Opentrons App. These instructions will help you get started:

> 1. Disconnect the robot's Ethernet cable from the wall jack and connect it to your computer.

Reviewer: The first instruction here assumes some odd context. Maybe the Flex isn't connected to anything when they start this! This will integrate these steps better within the section:
```suggestion
You can get the IP address range and subnet mask from the robot by connecting it to your computer and checking the Opentrons App:

1. If the robot is connected by Ethernet cable to a switch or wall jack, disconnect it. Then establish a physical Ethernet connection to your computer, as described above.
```

1. Launch the Opentrons App.

> 1. Click the Devices tab and find your robot.

Reviewer: ```suggestion
1. Click the **Devices** tab and find your robot.
```

    !!!note

>         If your robot appears as inactive or inaccessible in the app, wait a few moments. Flex will configure itself and eventually become available again. If this does not happen, turn the power off, wait a few seconds, turn the power back on, and check the app again after the robot boots up.

Reviewer: Clarifying which device to power cycle.
```suggestion
        If your robot appears as inactive or inaccessible in the app, wait a few moments. Flex will configure itself and eventually become available again. If this does not happen, turn the robot's power off, wait a few seconds, turn the power back on, and check the app again after the robot boots up.
```
